src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs

```diff
@@ -1806,6 +1841,31 @@ public override void VisitDiscardOperation(IDiscardOperation operation)
             LogCommonPropertiesAndNewLine(operation);
         }
 
+        public override void VisitDiscardPattern(IDiscardPatternOperation operation)
+        {
+            LogString(nameof(IDiscardPatternOperation));
+            LogPatternPropertiesAndNewLine(operation);
+        }
+
+        public override void VisitSwitchExpression(ISwitchExpressionOperation operation)
+        {
+            LogString($"{nameof(ISwitchExpressionOperation)} ({operation.Arms.Length} arms)");
+            LogNewLine();
+            Visit(operation.Value, nameof(operation.Value));
+            VisitArray(operation.Arms, nameof(operation.Arms), logElementCount: true);
+        }
+
+        public override void VisitSwitchExpressionArm(ISwitchExpressionArmOperation operation)
+        {
+            LogString($"{nameof(ISwitchExpressionArmOperation)} ({operation.Locals.Length} locals)");
+            LogNewLine();
+            Visit(operation.Pattern, nameof(operation.Pattern));
+            if (operation.Guard != null)
```

Comment:
>if (operation.Guard != null) [](start = 12, length = 28)

Is this conditional logic necessary?

Reply:
As it turns out, it makes the output much more readable for the most common scenarios.

---
In reply to: [244206611](https://github.com/dotnet/roslyn/pull/32018#discussion_r244206611) [](ancestors = 244206611)